test/linearforgettingweights.jl

```diff
@@ -0,0 +1,31 @@
+module TestLinearForgettingWeights
+
+using Compat
+import Distributions
+using Test
+
+using TreeParzen
+
+# just testing function itself
+
+# N is a number of observations/mus, lf is based on config and indicates a positive linear forgetting value beyond which
+# older points are de-weighted, e.g. if lf == 25, only values > 25 will be taken into consideration
+# case 1: when N == 0 is it possible to achieve, or can there never be the case when N == 0? what can be its use case:
+# no previous history?
+case1 = TreeParzen.LinearForgettingWeights.linear_forgetting_weights(0, 1)
+@test length(case1) == 0
+# case 2: when N < lf
+case2 = TreeParzen.LinearForgettingWeights.linear_forgetting_weights(6, 7)
+@test minimum(case2) == 1.0
+# case 3: when N - lf == 1
+case3 = TreeParzen.LinearForgettingWeights.linear_forgetting_weights(8, 7)
```

Comment:
I er ... feel like this is wrong?

So we have 8 observations and a linear forgetting history of 7

Linear forgetting history (I believe) is meant to make everything not older than its size as 1's
Everything older than its size goes down a linear ramp towards 0?
Thus, if everything as I have written so far is right, then if the history size is 7 and observation size is 8, final observation should be 0.5, no?

Reply:
@iqml can you remark?

Reply:
There seems to be a special case `if N - lf == 1`

Reply:
Ah, because `range` with `length = 1` doesn't make sense in this context.

Reply:
ok: I can see where my fundamental misunderstanding is .... 
The function is named "linear forgetting weights" which makes me believe the ramp should be linear but it is not, it is in fact inversely propotional to history age

Reply:
Should there be  a test to check that the number of 1 weights is equal to lf? At the moment we are only checking the minima.

Reply:
Let's rename the function

Reply:
any good suggestions? `non-linear` might be misleading...

Reply:
I removed the `linear` reference from its name and kept just `forgettingweights` for now.